lib/image-stream/utils.js

```diff
@@ -16,47 +16,30 @@
 
 'use strict';
 
-const _ = require('lodash');
 const Bluebird = require('bluebird');
-const fs = Bluebird.promisifyAll(require('fs'));
-const archiveType = require('archive-type');
+const mime = require('mime-types');
 
 /**
  * @summary Get archive mime type
  * @function
  * @public
  *
- * @param {String} file - file path
+ * @param {String} filename - file path
  * @fulfil {String} - mime type
  * @returns {Promise}
  *
  * @example
- * utils.getArchiveMimeType('path/to/raspberrypi.img.gz').then((mimeType) => {
- *   console.log(mimeType);
- * });
+ * const mimeType = utils.getArchiveMimeType('path/to/raspberrypi.img.gz');
  */
-exports.getArchiveMimeType = (file) => {
-
-  // `archive-type` only needs the first 261 bytes
-  // See https://github.com/kevva/archive-type
-  const ARCHIVE_TYPE_IDENTIFICATION_BYTES_LENGTH = 261;
-
-  return Bluebird.using(fs.openAsync(file, 'r').disposer((fileDescriptor) => {
-    return fs.closeAsync(fileDescriptor);
-  }), (fileDescriptor) => {
-    const BUFFER_START = 0;
-    const chunk = new Buffer(ARCHIVE_TYPE_IDENTIFICATION_BYTES_LENGTH);
+exports.getArchiveMimeType = (filename) => {
+  // NOTE: For some awkward reason `mime-db` (used by `mime-types`)
+  // does not contain a file extension for `.gz`, so we need to test for it here
+  // TODO: Remove this once https://github.com/jshttp/mime-db/pull/68 is merged
+  if (/\.gz$/i.test(filename)) {
+    return 'application/gzip';
+  }
 
-    return fs.readAsync(
-      fileDescriptor,
-      chunk,
-      BUFFER_START,
-      ARCHIVE_TYPE_IDENTIFICATION_BYTES_LENGTH,
-      null
-    ).then(() => {
-      return _.get(archiveType(chunk), [ 'mime' ], 'application/octet-stream');
-    });
-  });
+  return mime.lookup(filename) || 'application/octet-stream';
```

Comment:
I don't see the benefit in using `archive-type` as it's just an intermediate dependency, and the same can be done in 3 LOC:

```js
var type = fileType(buf)
var exts = [ '7z', 'zip', ... ]
return exts.contains( type ) ? type : false
```

Plus, we don't even need to narrow it down like `archive-type` does, because image-stream checks for a handlers existence based on the mime type.

> whereas file-type needs the first 4100 bytes, but can identify a much larger range of file types.

Where does it say that? Looking at the code, the highest offset into the buffer I could find is 261.

> and given that restriction (i.e. that archive-type doesn't even attempt to identify DMGs)

It doesn't, but as outlined earlier, it will detect other file types incorrectly *instead of* a DMG, because DMGs can contain anything in the first couple of bytes as they don't have any data structures or magic numbers there.
So if we use `file-type` before `mime-types`, then it will incorrectly detect a DMG as something else.